src/services/appServices.ts

```diff
@@ -0,0 +1,61 @@
+import UserModal from '../Models/user';
+import Board, { List } from '../types/board';
+import BoardModel, { List as ListModel } from '../Models/board';
+
+// export const storeBoard = async (userId: string, data: Board) => {
+//   try {
+//     const board = new BoardModel({
+//       title: data.title,
+//       isArchived: data.isArchived,
+//       lists: data.lists,
+//     });
+
+//     await board.save();
+//     const user =
+
+//     return {
+//       status: '201',
+//       message: 'Board created successfully',
+//     };
+
+//   } catch (err) {
+//     console.log(err);
+
+//     return {
+//       status: '400',
+//       message: 'There was an error',
+//     };
+//   }
+// };
+
+// export const getBoards = async () => {
+//   const result = await BoardModel.find().select(['title', '_id', 'isArchived', 'createdOn', 'lists']);
+//   return result;
+// };
+
+export const storeList = async (data: any) => {
+  const result = (await BoardModel.findById(data.params.id).select('lists')) as any;
+  const updatedList = [...result.lists, { name: data.body.name, level: data.body.level }];
+  result.lists = updatedList;
+  console.log(result);
+  result.save();
```

Comment:
It's better to separate the DAO layer in this case. Reasons:
- Remove code redundancy
- Isolate the application/business layer from the persistence layer (database)

Example:

```ts
// list.service.ts
import * as listDAO from './dao/list.dao'

export async function store(data: ListPayload): ListInterface {
    const  listId = await listDAO.store<number>(data) 
    const list = await listDAO.findById<T>(listId)

    return data;
}
```

```ts
// list.dao.ts
import ListModel from './model/list.dao'

const TABLE = 'lists'

export async function store<T>(data: object): T {
    const result = await BoardModel.findById(data.params.id).select(TABLE)
    const list = [...result.lists, name: data.body.name, level: data.body.level];
    const listId = result.save();

    return listId;
}
```